### PR TITLE
Rewrite App initialization using async

### PR DESCRIPTION
**Pull request by martinpitt**

### CODE REVIEW

File: src/app.jsx [hunk 1]
```diff
@@ -82,41 +77,37 @@ export const App = () => {
...
+            await initState();
+            const connectionNames = await unknownConnectionName();
+
+            await Promise.allSettled(connectionNames.map(async connectionName => {
+                try {
+                    await getLibvirtVersion({ connectionName });
+                    const promises = await getApiData({ connectionName });
+                    const errorMsgs = promises
+                            .filter(promise => promise.status === 'rejected')
+                            .map(promise => promise.reason.message);
```
Comment (cockpituous) [inline src/app.jsx:90]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1404-20240125-095117-bce4fcf9-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (cockpituous) [inline src/app.jsx:90]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1404-20240129-100518-70521786-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (cockpituous) [inline src/app.jsx:90]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1404-20240129-121536-2bf5d707-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (cockpituous) [inline src/app.jsx:90]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1404-20240129-134001-6e94a503-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/app.jsx [hunk 2]
```diff
@@ -179,28 +170,25 @@
...
-                            .then(() => this.setState({ cloudInitSupported: true }),
-                                  () => this.setState({ cloudInitSupported: false }));
-
-                    cockpit.spawn(['virt-install', '--unattended=?'], { err: 'ignore' })
-                            .then(options => this.setState({ unattendedSupported: true, unattendedUserLogin: options.includes('user-login') }),
-                                  () => this.setState({ unattendedSupported: false }));
-                },
-                      () => this.setState({ virtInstallAvailable: false }));
+        const check_exec = argv => cockpit.spawn(argv, { err: 'ignore' })
+                .catch(() => false);
```
Comment (cockpituous) [inline src/app.jsx:180]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1404-20240129-134001-6e94a503-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]

File: src/app.jsx [hunk 3]
```diff
@@ -82,41 +77,37 @@ export const App = () => {
...
-                    }));
-                })
-                .finally(() => setLoadingResources(false));
+        (async () => {
+            await initState();
+            const connectionNames = await unknownConnectionName();
+
+            await Promise.allSettled(connectionNames.map(async connectionName => {
+                try {
+                    await getLibvirtVersion({ connectionName });
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply src/app.jsx:86]: No, it's fine. Pretty much everything in c-machines puts state and results into redux, this as well:
```js
export async function getLibvirtVersion({ connectionName }) {
    const [version] = await call(connectionName, "/org/libvirt/QEMU", "org.freedesktop.DBus.Properties", "Get", ["org.libvirt.Connect", "LibVersion"], { timeout, type: "ss" });
    store.dispatch(updateLibvirtVersion({ libvirtVersion: version.v }));
}
```

[the remark this answers is not in the document]
Comment (jelly) [reply src/app.jsx:86]: Aaaah, guess I could confused the function name is a getter yet it get's nothing more a `storeLibvirtVersion` :)

**Pull request merged**